packages/datadog-instrumentations/src/router.js

```diff
@@ -169,11 +169,100 @@ function createWrapRouterMethod (name) {
 
 const wrapRouterMethod = createWrapRouterMethod('router')
 
-addHook({ name: 'router', versions: ['>=1'] }, Router => {
+addHook({ name: 'router', versions: ['>=1 <2'] }, Router => {
   shimmer.wrap(Router.prototype, 'use', wrapRouterMethod)
   shimmer.wrap(Router.prototype, 'route', wrapRouterMethod)
 
   return Router
 })
 
+addHook({ name: 'router', versions: ['>=2'] }, Router => {
+  const originalRouter = Router
+
+  function WrappedRouter (options) {
+    const router = originalRouter.call(this, options)
+
+    if (!router._queryParsingWrapped) {
+      router._queryParsingWrapped = true
+      const originalHandle = router.handle
+      router.handle = function (req, res, next) {
+        req.query
+        // If query parsing was aborted, don't continue request handling
+        if (res.writableEnded) {
+          return
+        }
```

Comment:
this is a bit hacky and feature related, I think that we should send a custom event in diagnostic_channels in this point, call to the waf and use the AbortController to decide if we should continue or not.